libs/cdk/utils/functions/lodash-utils.ts

```diff
@@ -0,0 +1,302 @@
+/**
```

Comment:
do we need to name it lodash if it is our own functionality?

Reply:
no, we can named it anything
but I think for the short term it's good to be called lodash so if a bug raised and someone took a look they can compare how the new function behave compared to the lodash library